### DISCUSSION TITLE
`--changed-within` with an absolute file path?

### DISCUSSION THREAD

**Discussion by xfzv** [sharkdp/fd#1579] [Q&A] [answered]
```console
% uname -o
GNU/Linux

% fd -V
fd 10.1.0
```

I'm currently using `find` in a Bash script to determine if a file was modified less than 60 minutes ago:

```console
output="/path/to/filename.ext"
% find "$output" -mmin +60
/path/to/filename.ext
```

`fd` doesn't print anything:

```console
output="/path/to/filename.ext"
% fd --changed-within=60min "$output"
% echo $?
0
```

This works:

```console
% fd --changed-within=60min filename.ext /path/
/path/to/filename.ext  
```

but I'd like to keep using `$output` (absolute file path) if possible. Just wondering if that's possible.

Thanks.

**Comment by tmccombs** [comment]
The first argument to `fd` is a pattern, not a path. 

You either need to do something like:

```
$ fd --changed-within=60min . "$output"
```

or 
```
$ fd --changed-within=60min --search-path "$output"
```

**Reply by xfzv** [reply]
Doesn't work:

```console
% output="/path/to/filename.ext"

% fd --changed-within=60min . "$output"
[fd error]: Search path '/path/to/filename.ext' is not a directory.
[fd error]: No valid search paths given.

% fd --changed-within=60min --search-path "$output"
[fd error]: Search path '/path/to/filename.ext' is not a directory.
[fd error]: No valid search paths given.
```

**Reply by tmccombs** [reply, marked as answer]
ah, yeah. 

This isn't really what fd was designed for. 

But you could do it with something like:

```bash
$ fd --changed-within=60min --fixed-strings "$(basename "$output")" "$(dirname "$output")"
```

**Reply by xfzv** [reply]
Nice trick with `basename` and `dirname`, I didn't think about it. Works!

Is `--fixed-strings` actually required though? It seems to also work without it, same with `--glob` instead. In my case, `filename.ext` only contains letters.

Thank you.

**Reply by tmccombs** [reply]
--fixed-strings is only necessary if your filename has special characters.

**Reply by xfzv** [reply]
Just as I thought, thank you for the confirmation.

**Reply by tavianator** [reply]
`--fixed-strings` is not perfect because it performs a substring match:

```console
tavianator@tachyon $ touch foo foofoo
tavianator@tachyon $ output=foo
tavianator@tachyon $ fd --changed-within=60min --fixed-strings "$(basename "$output")" "$(dirname "$output")"
./foo
./foofoo
```

`--glob` is not perfect either because of special characters:

```console
tavianator@tachyon $ touch 'a*a' abba
tavianator@tachyon $ output='a*a'
tavianator@tachyon $ fd --changed-within=60min --glob "$(basename "$output")" "$(dirname "$output")"
./a*a
./abba
```

If that matters, I think this works correctly in all cases:

```console
tavianator@tachyon $ fd --changed-within=60min --exact-depth=1 --glob "$(basename -- "$output" | sed 's/./\\&/g')" "$(dirname -- "$output")"
./a*a
```

Basically making `a*b` into the glob `\a\*\b`.
